js/controller.js

```diff
@@ -1452,14 +1452,14 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
 
     onChangeClosedCaptionLanguage: function(event, language) {
       var availableLanguages = this.state.closedCaptionOptions.availableLanguages;
-
       //validate language is available before update and save
-      if (language && availableLanguages && _.contains(availableLanguages.languages, language)) {
+        if (language && availableLanguages && (_.contains(availableLanguages.languages, language) || language === OO.CONSTANTS.CLOSED_CAPTIONS.NONE)) {
         this.state.closedCaptionOptions.language = this.state.persistentSettings.closedCaptionOptions.language = language;
-        var captionLanguage = this.state.closedCaptionOptions.enabled ? language : "";
-        var mode = this.state.closedCaptionOptions.enabled ? OO.CONSTANTS.CLOSED_CAPTIONS.HIDDEN : OO.CONSTANTS.CLOSED_CAPTIONS.DISABLED;
+        var captionLanguage = this.state.closedCaptionOptions.enabled && language !== OO.CONSTANTS.CLOSED_CAPTIONS.NONE ? language : "";
```

Comment:
In both of these cases, what is the value of captionLanguage and mode if the first variable is false? Is it a valid value?

Reply:
@pilievOoyala 
captionLanguage will have empty string if language === none.